src/hearings/containers/request-hearing/hearing-edit-summary/hearing-edit-summary.component.ts

```diff
@@ -444,49 +446,55 @@ export class HearingEditSummaryComponent extends RequestHearingPageFlow implemen
-    )?.filter(
-      (interpreterLanguage) => interpreterLanguage !== null && interpreterLanguage !== undefined
-    ).sort((a, b) => {
-      return a > b ? 1 : (a === b ? 0 : -1);
-    });
-    const interpreterLanguagesHMC = individualParties.map(
+    const returnPartyDetails = partyDetails.map((parties) => parties.partyID + ',' + parties.individualDetails?.reasonableAdjustments?.
+      filter((adjustments) => adjustments.startsWith('RA'))).
+      filter((adjustments) => !adjustments.includes(',undefined', null)).
+      filter((adjustments) => !adjustments.endsWith(','));
+    return returnPartyDetails;
+  }
+
+  private extractInterpreterLanguages(partyDetails: PartyDetailsModel[]) {
+    // Return true if there are changes to the interpreter languages
+    return partyDetails.map(
       (party) => party.individualDetails?.interpreterLanguage
     )?.filter(
       (interpreterLanguage) => interpreterLanguage !== null && interpreterLanguage !== undefined
     ).sort((a, b) => {
       return a > b ? 1 : (a === b ? 0 : -1);
     });
-    if (!_.isEqual(interpreterLanguagesSHV, interpreterLanguagesHMC)) {
-      return true;
-    }
+  }
 
-    // HMC stores only reasonable adjustment flag ids and language interpreter flag ids under parties
-    // Get only the reasonable adjustment and language interpreter flag ids from SHV and sort them for easy comparison
-    const caseFlagsSHV = this.serviceHearingValuesModel.caseFlags?.flags || [];
-    const flagIdsSHV = caseFlagsSHV.map((flag) => flag.flagId)?.filter((flagId) => flagId?.startsWith('RA'))?.sort((a, b) => {
-      return a > b ? 1 : (a === b ? 0 : -1);
-    });
-    // Get individual parties reasonable adjustment flags and sort the result
-    const partyFlagIds = individualParties.map((party) => party.individualDetails?.reasonableAdjustments)?.join(',').split(',').sort((a, b) => {
-      return a > b ? 1 : (a === b ? 0 : -1);
+  private extractAndSortParties(partyDetails: PartyDetailsModel[]){
+    // Get the individual parties
+    let individualParties = cloneDeep(partyDetails);
+    individualParties = individualParties.filter(
+      (party) => party.partyType === PartyType.IND).sort((a, b) => {
+      return a.partyID > b.partyID ? 1 : (a.partyID === b.partyID ? 0 : -1);
     });
-    // Convert to string for easy comparison as the partyFlagIds can be an array of empty strings;
-    const partyFlagIdsString = partyFlagIds?.length > 0
-      ? partyFlagIds?.filter((flagId) => flagId !== '').join()
-      : '';
-
-    // Return true if there are changes in reasonable adjustments and/or language interpreter flags
-    if (flagIdsSHV.join() !== partyFlagIdsString) {
-      return true;
-    }
-    // There are no changes for reasonable adjustments and language interpreter flags when compared SHV with HMC
-    return false;
+    individualParties.forEach(
+      (party) => party.individualDetails?.reasonableAdjustments?.sort((a, b) => {
```

Comment:
Is there a reason you are using a special sort function here? It looks pretty standard - you should be able to use reasonableAdjustments?.sort()